biofefi/machine_learning/get_models.py

```diff
@@ -33,18 +62,21 @@ def get_models(
         if model_class := _MODEL_PROBLEM_DICT.get(
             (model.lower(), problem_type.lower())
         ):
-            if problem_type.lower() == ProblemTypes.Classification:
-                model_param = assert_model_param(
-                    model_class, model_param, logger=logger
-                )
-                model_param["class_weight"] = "balanced"
-                models[model] = model_class(**model_param)
-            else:
-                model_param = assert_model_param(
-                    model_class, model_param, logger=logger
-                )
-                models[model] = model_class(**model_param)
+            if model in [ModelNames.BRNNClassifier, ModelNames.BRNNRegressor]:
+                models[model] = model_class(problem_type=problem_type, **model_param)
 
+            else:
+                if problem_type.lower() == ProblemTypes.Classification:
+                    model_param = assert_model_param(
+                        model_class, model_param, logger=logger
+                    )
+                    model_param["class_weight"] = "balanced"
+                    models[model] = model_class(**model_param)
+                else:
+                    model_param = assert_model_param(
+                        model_class, model_param, logger=logger
+                    )
+                    models[model] = model_class(**model_param)
```

Comment:
This bit won't work as I've moved `get_models` function and removed the `assert_model_params` function. I saw a bunch of conflicts in the PR that you will need to resolve. 

Also once you've sorted the conflicts can you make it so you don't have this special check for the BRNN models. Remember we're trying to make all our models behave the same so BioFEFI is easily extensible with more models. They should all follow the scikit-learn API